cloud/circuitbreaker/metricer.go

```diff
@@ -116,27 +116,29 @@ func (w *window) Succeed() {
 
 // Fail records a failure in the current perPBucket.
 func (w *window) Fail() {
-	w.rw.Lock()
+	rwx := w.rw.RLocker()
+	rwx.Lock()
 	b := w.getBucket()
```

Comment:
If `getBucket` is a readonly op, it can be moved out of the critical zone.

Reply:
There might be concurrent read and write on w.buckets